### ISSUE TITLE
Question on labels plot

### ISSUE THREAD

**Issue by vantienpham** [ultralytics/yolov5#2265]
## ❔Question
Hi, this is the label.png extracted from my custom training.
![image](https://user-images.githubusercontent.com/25927039/108654769-f3278980-74fb-11eb-9475-fc2be073e6c0.png)

I have 2 questions:
1. What is the meaning of the top-right subplot? I think that may be the ground-truth bouding box of N instances. Is that right?
2. In the top-left subplot, how to modify class label from "0, 1, 2" to "person, car, dog"? please point out the related code, I guess it shoud relate to utils/plots.py

Many thanks for your help!

**Comment by glenn-jocher** [comment]
@pvtien96 the top left plot is class frequency. This plot displays class indices, not the class names. You should supply a names array for your custom dataset yaml file as the Train Custom Data Tutorial describes below. The top right plot is a snapshot of your labels in normalized height and width space.

## Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [ONNX and TorchScript Export](https://docs.ultralytics.com/yolov5/tutorials/model_export)
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Comment by vantienpham** [comment]
@glenn-jocher thanks for your explanation.
With top left plot, I supplied class names as in the tutorial. 
However, as you mentioned, this subplot displays class indices by `    ax[0].hist(c, bins=np.linspace(0, nc, nc + 1) - 0.5, rwidth=0.8)
`
I tried `ax[0].set_xticks(['person', 'car', 'dog'])` but unfortunately hist doesn't work with set_xticks.

**Comment by glenn-jocher** [comment]
@pvtien96 you can customize this plot to suit your needs here. We don't assign label names to the axis as this doesn't scale to larger datasets like COCO.
https://github.com/ultralytics/yolov5/blob/095d2c11d89892cd9c0c4d034cd1c768a0dba11c/utils/plots.py#L272-L273

**Comment by vantienpham** [comment]
Thank you very much @glenn-jocher .

**Comment by vantienpham** [comment]
Just realize that the label plot now has name of classes on it. So happy :)
![image](https://user-images.githubusercontent.com/25927039/115369839-b40f6f00-a1f2-11eb-979c-65b417a3c996.png)

**Comment by glenn-jocher** [comment]
@pvtien96 yes we added this a while ago to improve the plots! Thanks for the feedback. If there are other feature requests or issues you encounter please let us know!

**Comment by vantienpham** [comment]
@glenn-jocher I'm wondering whether this repo should be released as an official paper. Your work contains a bunch of technique can be considered as contributions for the paper. Also, in many object detection competitions this yolov5 dominates the leader board. It is also used widely in other stuffs. One more things, I wrote in some papers that I used this repo and I cited this from your zenodo. But it'll be better if we have an arxiv/dbpl citation.
Should I open a feature request, haha?

**Comment by glenn-jocher** [comment]
@pvtien96 thanks for the suggestion! The paper is somewhere on our quite-long TODO list at the moment. We are working on bringing on some more maintainers for the repository and for expanding YOLOv5 training and deployment options as well, so hopefully soon I'll have some more free time to dedicate to tasks like the paper that I agree is quite overdue.
